### PR TITLE
PHX-161 added the new helper methods to create/update review policy for proctored exams

### PR DESCRIPTION
**Pull request by muhhshoaib**
@afzaledx can you do the internal review. 
FYI @chrisndodge

### CODE REVIEW

File: edx_proctoring/api.py [hunk 1]
```diff
@@ -96,6 +99,84 @@ def create_exam(course_id, content_id, exam_name, time_limit_mins, due_date=None
...
+def get_review_policy_by_exam_id(exam_id):
+    """
+    Looks up exam by the Primary Key. Raises exception if not found.
+
+    Returns dictionary version of the Django ORM object
+    e.g.
+    {
+        "id": 1
+        "proctored_exam": "{object}",
+        "user": "{object}",
```
Comment (afzaledx) [inline edx_proctoring/api.py]: set_by_user?

File: edx_proctoring/api.py [hunk 2]
```diff
@@ -96,6 +99,84 @@ def create_exam(course_id, content_id, exam_name, time_limit_mins, due_date=None
     return proctored_exam.id
 
 
+def create_exam_review_policy(exam_id=None, set_by_user_id=None, review_policy=None):
+    """
+    Creates a new exam_review_policy entity, if the review_policy for exam_id do not already exist.
```
Comment (afzaledx) [inline edx_proctoring/api.py]: exam_id does not

File: edx_proctoring/api.py [hunk 3]
```diff
@@ -96,6 +99,84 @@ def create_exam(course_id, content_id, exam_name, time_limit_mins, due_date=None
...
+    If already exists, then raise exception.
```
Comment (afzaledx) [inline edx_proctoring/api.py]: If it exists

File: edx_proctoring/api.py [hunk 4]
```diff
@@ -96,6 +99,86 @@ def create_exam(course_id, content_id, exam_name, time_limit_mins, due_date=None
...
+            review_policy=review_policy,
+            set_by_user_id=set_by_user_id
+        )
+    )
+    log.info(log_msg)
+
+    return exam_review_policy.id
+
+
+def update_review_policy(exam_id=None, set_by_user_id=None, review_policy=None):
```
Comment (afzaledx) [inline edx_proctoring/api.py]: All these arguments should be required. not optional.

File: edx_proctoring/api.py [hunk 5]
```diff
@@ -96,6 +99,86 @@ def create_exam(course_id, content_id, exam_name, time_limit_mins, due_date=None
...
+            exam_id=exam_id, set_by_user_id=set_by_user_id, review_policy=review_policy,
+        )
+    )
+    log.info(log_msg)
+    exam_review_policy = ProctoredExamReviewPolicy.get_review_policy_for_exam(exam_id)
+    if exam_review_policy is None:
+        raise ProctoredExamReviewPolicyNotFoundException
+
+    exam_review_policy.set_by_user_id = set_by_user_id
+    exam_review_policy.proctored_exam_id = exam_id
```
Comment (afzaledx) [inline edx_proctoring/api.py]: Can omit this.

**Comment by afzaledx** [comment]
:+1:

**Comment by muhhshoaib** [comment]
I had to rebase as there were duplicate commits in the PR.

**Comment by chrisndodge** [comment]
rebased against Django 1.8 merge

**Comment by chrisndodge** [comment]
Hi @muhhshoaib just a note on this PR. I think I'm going to hold off on this one until we finish our 2 upcoming proctored exams (dec.1 - dec.4). Thanks

**Comment by chrisndodge** [comment]
rebased this to pick up all changes from master

**Comment by chrisndodge** [comment]
@muhhshoaib given that this PR is totally additive (new methods, exception classes, etc.) we can merge this in while the corresponding PR in edx-platform gets reviewed by the UX team. I'll squash commits and merge when we get a clean build.

**Pull request merged**